source/developers-guide/legacy-plugin-system/index.md

```diff
@@ -0,0 +1,831 @@
+
+```php
+public function install()
+{
+    // ...
+    
+    $this->subscribeEvent(
+        'Enlight_Controller_Action_PostDispatchSecure_Backend_Emotion',
+        'onPostDispatchBackendEmotion'
+    );
+}
+
+public function onPostDispatchBackendEmotion(Enlight_Controller_ActionEventArgs $args)
+{
+    $controller = $args->getSubject();
+    $view = $controller->View();
+
+    $view->addTemplateDir($this->Path() . 'Views/');
+    $view->extendsTemplate('backend/emotion/vimeo_element/view/detail/elements/vimeo_video.js');
+}
+```
+Now that we added our template file we can add our custom component by extending the Smarty {block} of the base class.
+
+```js
+//
+//{block name="backend/emotion/view/detail/elements/base"}
+//{$smarty.block.parent}
+Ext.define('Shopware.apps.Emotion.view.detail.elements.VimeoVideo', {
+
+    extend: 'Shopware.apps.Emotion.view.detail.elements.Base',
+
+    alias: 'widget.detail-element-emotion-components-vimeo',
+
+    componentCls: 'emotion--vimeo-video',
+
+    icon: 'data:image/png;base64,...',
+
+    createPreview: function () {
+        var me = this,
+            preview = '',
+            image = me.getConfigValue('vimeo_video_thumbnail'),
+            style;
+
+        if (Ext.isDefined(image)) {
+            style = Ext.String.format('background-image: url([0]);', image);
+
+            preview = Ext.String.format('<div class="x-emotion-banner-element-preview" style="[0]"></div>', style);
+        }
+
+        return preview;
+    }
+});
+//{/block}
+```
+
+## Add a new payment method
+Use the "createPayment" method to add payment methods to the database inside plugin installations.
+
+```php
+public function install(InstallContext $context)
+{
+    $this->createPayment(
+        array(
+            'name' => 'payment name',
+            'description' => 'Pay by the new Payment',
+            'action' => 'payment_paymentName',
+            'active' => 0,
+            'position' => 0,
+            'additionalDescription' => 'Lorem ipsum ... ',
+        )
+    );
+}
+```
+
+### Plugin Configuration / Forms
+
+Backend plugin configuration can be extended by the usage of `$this->Form()->setElement()` in the install method of the plugin Bootstrap.php. 
+
+```php
+$form = $this->Form();
+
+// Now we can use this instance of Shopware\Models\Config\Form in $form to add elements to it via `setElement().
+
+$form->setElement(
+    
+    'color', 
+    'yourSetting',
+    [
+        'label' => 'Your label',
+        'description' => Lorem ipsum ...',
+        'value' => '#ffffff',
+        'scope' => Element::SCOPE_SHOP,
+    ]
+);
+```
+
+A textfield would be defined as followed:
+```php
+public function createConfiguration()
+{
+    $form = $this->Form();
+ 
+    $form->setElement(
+        'text', 
+        'simpleTextField',
+        array(
```

Comment:
the code style is still different. you use long array syntax here. above is an empty line after setElement(